File: LC367.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        int l = 1, h = num;

        if(num == 1)
            return true;

        while(l <= h){
            int m = l + (h-l)/2;

            if(1ll*m*m == num)
                return true;

            if(1ll*m*m > num)
                h = m-1;
            else
                l = m + 1;
        }

        return false;
    }
};

File: LC367.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num == 1:
            return num
        
        l = 1
        h = num//2

        while l <= h:
            m = (l+h)//2

            if m*m == num:
                return True
            
            if m*m > num:
                h = m - 1
            else:
                l = m + 1
        
        return False
